### PR TITLE
regression tests

### PR DESCRIPTION
**Pull request by FadyEssam**

### CODE REVIEW

File: modules/augment/test/augmenter_images.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    cv.setRNGSeed(seed=1)
+
+    numberOfImages = np.random.randint(10,100)
+
+    aug = cv.augment_Augmenter()
+    aug.add(cv.augment_FlipHorizontal(), 0.7)
+    aug.add(cv.augment_FlipVertical(), 0.5)
+    aug.add(cv.augment_GaussianBlur(5,12), .7)
+    aug.add(cv.augment_Rotate(0,180), 0.3)
+    aug.add(cv.augment_Resize((1200,900)),0.4)
```
Comment (dkurt) [inline modules/augment/test/augmenter_images.py]: Please use named parameters: param=value

File: modules/augment/test/augmenter_images.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+try:
+    import cv2 as cv
+except ImportError:
+    raise ImportError('Can\'t find OpenCV Python module. If you\'ve built it from sources without installation, '
+                      'configure environment variable PYTHONPATH to "opencv_build_dir/lib" directory (with "python3" subdirectory if required)')
```
Comment (dkurt) [inline modules/augment/test/augmenter_images.py]: Kind of redundant exception message: import cv2 already produces enough information

File: modules/augment/test/augmenter_images.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+try:
+    import numpy as np
+except ImportError:
+    raise ImportError('Can\'t find numpy library, please make sure it is installed')
```
Comment (dkurt) [inline modules/augment/test/augmenter_images.py]: Just import. No need to check

**Comment by dkurt** [comment]
Please use existing tests for reference. They should be a part of testing module. See https://github.com/opencv/opencv/blob/master/modules/dnn/misc/python/test/test_dnn.py for reference.

File: modules/augment/test/augmenter_images.py (hunk 3 above)
Comment (FadyEssam) [reply modules/augment/test/augmenter_images.py]: done

File: modules/augment/test/augmenter_images.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    cv.setRNGSeed(seed=1)
+
+    numberOfImages = np.random.randint(10,100)
+
+    aug = cv.augment_Augmenter()
+    aug.add(t=cv.augment_FlipHorizontal(), prob=0.7)
+    aug.add(t=cv.augment_FlipVertical(), prob=0.5)
+    aug.add(t=cv.augment_GaussianBlur(5,12), prob=0.7)
+    aug.add(t=cv.augment_Rotate(0,180), prob=0.3)
+    aug.add(t=cv.augment_Resize((1200,900)), prob=0.4)
```
Comment (dkurt) [inline modules/augment/test/augmenter_images.py]: Please add parameters names for all the numbers. It's a test and users will used it to understand an API.

File: modules/augment/misc/python/test/test_augmenter_images.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+    def test_augmenter_images_with_rectangles(self):
+        numberOfImages = np.random.randint(low=MIN_NUMBER_OF_TESTS, high=MAX_NUMBER_OF_TESTS)
+
+        aug = cv.augment_Augmenter()
+        aug.add(t=cv.augment_FlipHorizontal(), prob=0.7)
+        aug.add(t=cv.augment_FlipVertical(), prob=0.5)
+        aug.add(t=cv.augment_GaussianBlur(kernelSize=5, sigma=12), prob=0.7)
+        aug.add(t=cv.augment_Rotate(minAngle=0, maxAngle=180), prob=0.3)
+        aug.add(t=cv.augment_Resize(size=(1200, 900)), prob=0.4)
```
Comment (dkurt) [inline modules/augment/misc/python/test/test_augmenter_images.py:91]: When I asked about parameters names I meant only parameters which make sense to be names. Please do not try to name every parameter:

```python

        numberOfImages = np.random.randint(MIN_NUMBER_OF_TESTS, MAX_NUMBER_OF_TESTS)

        aug = cv.augment_Augmenter()
        aug.add(cv.augment_FlipHorizontal(), prob=0.7)
        aug.add(cv.augment_FlipVertical(), prob=0.5)
        aug.add(cv.augment_GaussianBlur(kernelSize=5, sigma=12), prob=0.7)
        aug.add(cv.augment_Rotate(minAngle=0, maxAngle=180), prob=0.3)
        aug.add(cv.augment_Resize(size=(1200, 900)), prob=0.4)
```

File: modules/augment/misc/python/test/test_augmenter_images.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+        for i in range(numberOfImages):
+            widthOfImages = np.random.randint(low=MIN_IMAGE_DIM_SIZE, high=MAX_IMAGE_DIM_SIZE)
+            heightOfImages = np.random.randint(low=MIN_IMAGE_DIM_SIZE, high=MAX_IMAGE_DIM_SIZE)
+            numberOfRects = np.random.randint(low=MIN_NUMBER_OF_GROUND_TRUTH_DATA, high=MAX_NUMBER_OF_GROUND_TRUTH_DATA)
+
+            img = np.random.rand(heightOfImages, widthOfImages)
+            imgs.append(img)
+            rects = np.random.rand(numberOfRects, 4)
+            rectsArr.append(rects)
```
Comment (dkurt) [inline modules/augment/misc/python/test/test_augmenter_images.py:104]: We need to specify which actual types of images and rectangles are allowed. So you need to show that images must be numpy arrays and rectangles are lists or tuples or numpy arrays.

Comment (FadyEssam) [reply modules/augment/misc/python/test/test_augmenter_images.py:104]: How can I make this more clear? should I write that in comments?

**Pull request merged**